libs/core/src/graphql/fragment.integration.spec.ts

```diff
@@ -0,0 +1,89 @@
+import { TestBed } from '@angular/core/testing';
+import { ApolloTestingController, ApolloTestingModule } from 'apollo-angular/testing';
+import { DocumentNode } from 'graphql';
+import gql from 'graphql-tag';
+import { Apollo } from 'apollo-angular';
+
+import { daffBuildFragmentSpread } from './build-fragment-spread';
+import { daffGetFragmentNames } from './get-fragment-names';
+
+describe('Core | GraphQL | Fragment Integration', () => {
+  let controller: ApolloTestingController;
+  let apollo: Apollo;
+
+  let mockFragment: DocumentNode;
+  let mockResponse;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        ApolloTestingModule
+      ]
+    });
+
+    controller = TestBed.get(ApolloTestingController);
+    apollo = TestBed.get(Apollo);
+
+    mockFragment = gql`
+      fragment fragment1 on Query {
+        field1
+      }
+      fragment fragment2 on Query {
+        field2
+      }
+    `;
+    mockResponse = {
+      id: 'id',
+      field1: 'field1',
+      field2: 'field2',
+    }
+  });
+
+  describe('adding fields to a query', () => {
+    let query;
+
+    beforeEach(() => {
+      query = gql`
+        query {
+          id
+          ${daffBuildFragmentSpread(daffGetFragmentNames(mockFragment))}
```

Comment:
I think this API is a bit clunky, I'd like to only have to pass a mockFragment, not a nested function.

Reply:
changed, see integration tests for final usage